web/web.go

```diff
@@ -0,0 +1,694 @@
+		return
+	}
+	// Filter typeMap so it contains just the types SrcSchema uses.
+	filteredTypeMap := make(map[string][]typeIssue)
+	for _, srcTable := range app.conv.SrcSchema {
+		for _, colDef := range srcTable.ColDefs {
+			if _, ok := filteredTypeMap[colDef.Type.Name]; ok {
+				continue
+			}
+			filteredTypeMap[colDef.Type.Name] = typeMap[colDef.Type.Name]
+		}
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(filteredTypeMap)
+}
+
+func setTypeMapGlobal(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var t map[string]string
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+
+	for k, v := range app.conv.SpSchema {
+		for kk, _ := range v.ColDefs {
+			for tk, tv := range t {
+				sourceTable := app.conv.ToSource[k].Name
+				sourceCol := app.conv.ToSource[k].Cols[kk]
+				srcCol := app.conv.SrcSchema[sourceTable].ColDefs[sourceCol]
+				if srcCol.Type.Name == tk {
+					var ty ddl.Type
+					var issues []internal.SchemaIssue
+					switch app.driver {
+					case "mysql", "mysqldump":
+						ty, issues = toSpannerTypeMySQL(srcCol.Type.Name, tv, srcCol.Type.Mods)
+					case "pg_dump", "postgres":
+						ty, issues = toSpannerTypePostgres(srcCol.Type.Name, tv, srcCol.Type.Mods)
+					default:
+						http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+						return
+					}
+					if len(srcCol.Type.ArrayBounds) > 1 {
+						ty = ddl.Type{Name: ddl.String, Len: ddl.MaxLength}
+						issues = append(issues, internal.MultiDimensionalArray)
+					}
+					if srcCol.Ignored.Default {
+						issues = append(issues, internal.DefaultValue)
+					}
+					if srcCol.Ignored.AutoIncrement {
+						issues = append(issues, internal.AutoIncrement)
+					}
+					if len(issues) > 0 {
+						app.conv.Issues[sourceTable][srcCol.Name] = issues
+					}
+					ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+					spSchema := app.conv.SpSchema[k]
+					colDef := spSchema.ColDefs[kk]
+					colDef.T = ty
+					spSchema.ColDefs[kk] = colDef
+					app.conv.SpSchema[k] = spSchema
+				}
+			}
+		}
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+func remove(slice []string, s int) []string {
+	return append(slice[:s], slice[s+1:]...)
+}
+func removePk(slice []ddl.IndexKey, s int) []ddl.IndexKey {
+	return append(slice[:s], slice[s+1:]...)
+}
+
+func removeColumn(table string, colName string, srcTableName string) {
+	sp := app.conv.SpSchema[table]
+	for i, col := range sp.ColNames {
+		if col == colName {
+			sp.ColNames = remove(sp.ColNames, i)
+			break
+		}
+	}
+	if _, found := sp.ColDefs[colName]; found {
+		delete(sp.ColDefs, colName)
```

Comment:
Its save to just call delete(sp.ColDefs, colName).
You don't need to check the value is there.

From godoc (https://golang.org/pkg/builtin/#delete):

func delete(m map[Type]Type1, key Type)

The delete built-in function deletes the element with the specified key (m[key]) from the map. If m is nil or there is no such element, delete is a no-op.

Reply:
removed extra check